### PR TITLE
issue-292 Autoblocking of breadcrumb block if its not present

### LINKED ISSUE #292: Automatically inject breadcrumbs on pages
Breadcrumbs are present across all pages. So its an unnecessary overhead for author to explicitly add them to each page and it introduces chances of manual error if the author fails to add it or add it incorrectly. Ideally the breadcrumb block should be automatically added to a page. For cases where its already present, the automatic addition can be skipped.

### PR DESCRIPTION
**Pull request by jindaliiita**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fixes #292 

Test URLs:
- Original: https://www.sunstar.com/healthy-thinking/4944/
- Before: https://main--sunstar--hlxsites.hlx.live/_drafts/piyush/4944
- After: https://auto-breadcrumb--sunstar--hlxsites.hlx.live/_drafts/piyush/4944

- [ ] New Blocks introduced in this PR
      If yes, please provide details below
**Block Name** : For e.g. cards
- [ ] Documented in sidekick library

- [ ] New variations introduced in this PR
**Variation Name** :  For e.g. cards (grid)
- [ ] Documented in sidekick library

- [ ] New mixins introduced in this PR
**Mixin Name** :  For e.g. compact, white
- [ ] Documented in sidekick library

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -180,12 +180,31 @@ export function buildImageWithCaptionBlocks(main, buildBlockFunction) {
...
+
+  if ((!noBreadcrumb || noBreadcrumb === 'false') && !alreadyBreadcrumb && !isInternalPage()) {
+    const section = document.createElement('div');
+    const blockEl = buildBlock('breadcrumb', { elems: [] });
+    section.append(blockEl);
+    main.prepend(section);
+  }
+}
+
+// doc.querySelector('.breadcrumb')
```
Comment (dnbute) [inline scripts/scripts.js:199]: Don't forget to delete this before merge

**Review by dnbute — COMMENTED** [review]
I feel like it will be weird to have 90% of pages with the breadcrumbs authored and a small amount of pages that have this autoblocked, should we go back and reauthor the pages with the breadcrumbs out? Might be worth discussing with the team

**Comment by dnbute** [comment]
> > I feel like it will be weird to have 90% of pages with the breadcrumbs authored and a small amount of pages that have this autoblocked, should we go back and reauthor the pages with the breadcrumbs out? Might be worth discussing with the team
> 
> Idea was to not add breadcrumb on each and every page. We will not make any change on the pages which are already authored.

Yes, exactly, not have breadcrumb on every page in the doc, we can use something like @bosschaert's [docxtools](https://github.com/coderthoughts/docxtools) (please correct me if I'm wrong) to mass delete the `breadcrumb` on the currently authored pages, it's a bit of an effort for sure, but I think in the long run having all pages as uniform as possible will be worth it

**Review by JiangLong2019 — APPROVED** [review]

**Comment by jindaliiita** [comment]
> > > I feel like it will be weird to have 90% of pages with the breadcrumbs authored and a small amount of pages that have this autoblocked, should we go back and reauthor the pages with the breadcrumbs out? Might be worth discussing with the team
> > 
> > 
> > Idea was to not add breadcrumb on each and every page. We will not make any change on the pages which are already authored.
> 
> Yes, exactly, not have breadcrumb on every page in the doc, we can use something like @bosschaert's [docxtools](https://github.com/coderthoughts/docxtools) (please correct me if I'm wrong) to mass delete the `breadcrumb` on the currently authored pages, it's a bit of an effort for sure, but I think in the long run having all pages as uniform as possible will be worth it @sdmcraft thoughts?

I thinks we can follow the same process here
https://github.com/hlxsites/sunstar/pull/321#issuecomment-1808312698. WDYT ?

**Comment by sdmcraft** [comment]
Yes, this is a good simplification in the authoring experience and we should make authors aware about it. I'd suggest starting with deprecating the `breadcrumbs` block in block library, followed by removal from documents. I'd suggest logging a separate ticket for this.

**Review by sdmcraft — APPROVED** [review]

**Pull request merged**